### PR TITLE
Support demo env

### PR DESCRIPTION
**Pull request by joofsh**
This PR adds support for a `demo` environment, to be used on the current "hosted dev".  This gives us the benefits of a production-like setup with precompiled assets but with the faked out integrations.

Testing:
- Setting up demo database:
screen shot 2016-11-18 at 5 58 52 pm

- Running app in demo mode:
screen shot 2016-11-18 at 5 56 52 pm
screen shot 2016-11-18 at 6 01 26 pm

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -21,6 +21,17 @@ development_vacols:
   timeout: 5000
   database: db/development-vacols.sqlite3
 
+demo:
```
Comment (joofsh) [inline config/database.yml]: Exact copy of `development` above

File: config/environments/demo.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
+Rails.application.configure do
```
Comment (joofsh) [inline config/environments/demo.rb]: This is an exact copy of the `production.rb` with the exception of the GA code on line 78

File: config/environments/demo.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  # Enable Rack::Cache to put a simple HTTP cache in front of your application
+  # Add `rack-cache` to your Gemfile before enabling this.
+  # For large-scale production use, consider using a caching reverse proxy like
+  # NGINX, varnish or squid.
+  # config.action_dispatch.rack_cache = true
+
+  # Disable serving static files from the `/public` folder by default since
+  # Apache or NGINX already handles this.
+  config.serve_static_files = ENV['RAILS_SERVE_STATIC_FILES'].present?
```
Comment (joofsh) [inline config/environments/demo.rb]: Note: If we don't have nginx setup in the demo environment, we'll need to set `ENV['RAILS_SERVE_STATIC_FILES'] = true`

File: config/environments/demo.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  # Compress JavaScripts and CSS.
+  config.assets.js_compressor = :uglifier
+  # config.assets.css_compressor = :sass
+
+  # Do not fallback to assets pipeline if a precompiled asset is missed.
+  config.assets.compile = false
```
Comment (joofsh) [inline config/environments/demo.rb]: forces assets to the precompiled. The relevant difference from dev mode

File: config/routes.rb [hunk 5]
```diff
@@ -35,7 +35,7 @@
   patch "certifications" => "certifications#create"
 
   # :nocov:
-  if Rails.env.development? || Rails.env.test?
+  if Rails.env.development? || Rails.env.test? || Rails.env.demo?
```
Comment (joofsh) [inline config/routes.rb]: This could be `%w(demo development test).include?(Rails.env)` if you prefer?

**Comment by joofsh** [comment]
@askldjd @orischwartz Note [the logger setup](https://github.com/department-of-veterans-affairs/caseflow/blob/master/config/initializers/logger.rb#L22-L32) would be the one difference for hosted dev from how it current operates today. I don't see a problem with this but let me know if I'm mistaken

File: config/environments/demo.rb (hunk 4 above)
Comment (shanear) [inline config/environments/demo.rb]: 👍

File: config/routes.rb (hunk 5 above)
Comment (shanear) [inline config/routes.rb]: this is fine by me

**Comment by orischwartz** [comment]
:+1: LGTM, need a corresponding PR in deployment to pass the new `RAILS_ENV` value, specified via the [`rails_env` ansible variable here](https://github.com/department-of-veterans-affairs/appeals-deployment/blob/master/ansible/roles/efolder-express/vars/vault_dev.yml#L3).

File: config/database.yml [hunk 6]
```diff
@@ -21,6 +21,17 @@ development_vacols:
...
+  <<: *default
+  username: <%= `whoami` %>
+  database: caseflow_certification_demo
```
Comment (joofsh) [inline config/database.yml]: @orischwartz @askldjd One thing I forgot to bring up. The database name has been changed here (was `caseflow_certification_development`). Will this affect the initial deploy to hosted dev? I assume the deploy isn't running `rake db:create`? 

Alternatively I can name this development so it matches the existing db name

**Pull request merged**